buffer.py

```diff
@@ -902,6 +916,14 @@ def paintEvent(self, event):
         else:
             painter.setPen(inverted_color((self.theme_foreground_color)))
 
+        # Draw an indicate to synctex position
+        if self.synctex_pos_y and (self.synctex_pos_y > 0):
```

Comment:
synctex_pos_y should add page top y coordinate, become global coordinate, then you don't need call clear_synctex_indicator everywhere.

Reply:
Could you suggest how to get *page top y coordinate*?
Is it the`page_offset` variable in the function `get_cursor_absolute_position`?

I also face another problem is that when a page is zoomed too big, and the indicator is at the end of the page, then the indicator will not be visible in the widget. I haven't been able to find a way to scroll the page and draw the indicator at the correct location.

Reply:
IMO, when you click LaTex document, indicator should be draw at fixed position and don't matter page is zoom.

Indicator should hide when user click other place or cancel indicator.

You need read code of function paintEvent, I don't use LaTex, can't help you, sorry.